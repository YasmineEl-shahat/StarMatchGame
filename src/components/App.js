import * as React from 'react';

export function App({ initialData }) {
  const [count, setCount] = React.useState(0);
  return (
    <div>
      <h1>{initialData.appName}</h1>
      This is a sample stateful and server-side rendered React application.
      <br />
      <br />
      Here is a button that will track how many times you click it:
      <br />
      <br />
      <button title="increment" onClick={() => setCount(count + 1)}>
        {count}
      </button>
    </div>
  );
}
