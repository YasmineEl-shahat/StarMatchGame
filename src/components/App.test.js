import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { App } from './App';

import { render, fireEvent, screen } from '@testing-library/react';

describe('App', () => {
  it('renders ansd increments counter', () => {
    const { getByText, getByTitle, asFragment } = render(
      <App initialData={{ appName: 'TEST' }} />,
    );
    expect(getByText('TEST')).toMatchInlineSnapshot(`
      <h1>
        TEST
      </h1>
    `);

    const button = screen.getByTitle('increment');
    fireEvent.click(button);
    expect(getByTitle('increment')).toHaveTextContent('1');

    expect(asFragment()).toMatchSnapshot();
  });
});
