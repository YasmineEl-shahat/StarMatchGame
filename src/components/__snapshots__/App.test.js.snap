// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders ansd increments counter 2`] = `
<DocumentFragment>
  <div>
    <h1>
      TEST
    </h1>
    This is a sample stateful and server-side rendered React application.
    <br />
    <br />
    Here is a button that will track how many times you click it:
    <br />
    <br />
    <button
      title="increment"
    >
      1
    </button>
  </div>
</DocumentFragment>
`;
